### DISCUSSION TITLE
Using wait_for_build_processing_to_be_complete with API Key

### DISCUSSION THREAD

**Discussion by tiago-anchorage** [fastlane/fastlane#17927] [General]
I am trying to use the ```FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete``` method with an API Key.

I was able to use it with Spaceship::Tunes login like this:

```ruby
fastlane_require 'spaceship'

Spaceship::Tunes.login
Spaceship::Tunes.select_team

app = Spaceship::Tunes::Application.find(app_identifier)

FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete(
   app_id: app.apple_id, 
   app_version: '2.9.1', 
   build_version: '0'
)
```

However, I want to use the API key, So I have been trying to do something like this:

```ruby
token = Spaceship::ConnectAPI::Token.create(
    key_id: 'id',
    issuer_id: 'issuer_id',
    filepath:  File.absolute_path("path")
)

Spaceship::ConnectAPI.token = token
Spaceship::ConnectAPI.select_team
```

But I got this error 

>Please login using `Spaceship::Tunes.login('user', 'password')`

So I assume this doesnt work with App Key?

Basically what I am trying to do is wait for a build to process so I can download the dSYMs 😅

**Comment by rogerluan** [comment]
Hey @tiago-anchorage 👋 

Have you tried using [app_store_connect_api_key](https://docs.fastlane.tools/app-store-connect-api/#use-the-shared-value-in-lane-context)? Not saying that managing the token is wrong, but wondering if you tried other alternatives 🙈 I couldn't pin point the reason you're seeing that error by quickly glancing at the code

**Reply by tiago-anchorage** [reply]
I did, still with no success. 

I ended up using the parameter wait_for_dsym_processing in download_dsyms, but it would be cool if I knew how to go into "manual mode" if needed 😛

**Reply by rogerluan** [reply]
Hey @tiago-anchorage , I finally got some time to investigate this issue. In fact, if you use `wait_for_build_processing_to_be_complete`'s `return_spaceship_testflight_build: true` (which is the default), it doesn't support API key yet. However, this parameter only modifies the response, and it doesn't look like you're reading the method's response at all, you just want to wait for it. In that case you can pass:

```ruby
FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete(
  app_id: '123231321',
  app_version: '1.2.3',
  build_version: '123',
  return_spaceship_testflight_build: false
)
```

And it will work just fine. If you happen to use the output, the response type will be a Spaceship::ConnectAPI::Build instance, which's different than the response when using the old iTunes auth 👍 

Not sure if there're plans to support API key in this flow any time soon - it would require some refactoring otherwise I'd have done it just now 😅 

Hope this help you out!
